app/src/main/java/mollie/tictactoe/StateConverter.java

```diff
@@ -0,0 +1,66 @@
+package mollie.tictactoe;
+
+import mollie.tictactoe.players.MobilePlayer;
+import ttt.Player;
+import ttt.game.Board;
+import ttt.game.GameEngine;
+import ttt.game.Marks;
+
+public class StateConverter {
```

Comment:
StateConverter may have a couple of minor illnesses.
He suffers from a slight pulmonary string-oma. You have a great enum `Mark` with X and O. The strings should only be _outputs_ to the screen, but never inputs. If you have some callbacks (haven't checked) then just have them remember the `Mark`.

All the statics mights be another illness... If you look at them, they reach deep into other data structures (getCurrentMark) to then do a slight conversion. If your app used more Marks instead of Strings, you'd have to convert less..